### ISSUE TITLE
SignIn Missing Explicit Indication of Sign In Success or Failure

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#791]
When a user has successfully signed in using either Microsoft or Google, they are redirected back to their previous location without any clear indication of the fact that they are now signed in.  On the Landing Page, the “Sign In” button’s label is updated to “Go to App”, while on /app pages the navigation menu contains a user profile section.  Neither of these are sufficient as accessible indications of sign in success. 
 
No error paths are defined in the code to assist a user should the OAuth process be unsuccessful.  It is assumed that if a user profile object cannot be retrieved from the OAuth provider’s API, the Sign In page will stay on the screen without any error messages or other feedback to tell the user that something is wrong. 

Suggested Solution 
 
If sign in is successful, either: 
 
• Inform the user via the Sign In page, by showing a success message and setting keyboard focus to it.  For example: 
o Heading level 2: “Signed In” (should receive keyboard focus and therefore should have a tabindex attribute with a value of “-1”). 
o Text: “You are now signed in as [name]”. 
o Button: “Continue to MathShare Application” 

• Or, alternatively, redirect the user to their previous page, but explicitly indicate that they are now signed in.  A similar structure can be used as in the first option, with a heading which receives focus and some welcome text. 
 
If a sign in attempt is not successful, for example because the user’s profile cannot be retrieved from the remote API, users must be clearly informed of this fact.  As in the implementation for the happy path recommended above, the error message should have a heading which receives focus, some explanatory text about what went wrong, and any relevant controls such as “Retry”, “Continue as a Guest”, etc.  For example: 
 
• Heading level 2: “Sign In Failed” 
• Text: “Sorry, something went wrong. Your user profile could not be retrieved from Microsoft. You can try to sign in again or return to MathShare as a guest.”
• Button: “Try Again” 
• Link: “Return to MathShare as a guest”

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/3a0e32e6ed8d6b190286fc64c91aecd6f6a9d825.
